### DISCUSSION TITLE
Hack Language detected.

### DISCUSSION THREAD

**Discussion by Soumilgit** [github-linguist/linguist#6808] [Classification]
## Describe the bug
Hack language is also detected in my repository as well repo link:-https://github.com/Soumilgit/E-Commerce-WPL-My-work

### Expected behaviour
I know PHP and Hack are considered same by Github,still can the display issue be resolved?Also,why both are same?

### Related discussion
CLOSED ISSUE:-https://github.com/github-linguist/linguist/issues/4229

![Hacklang](https://github.com/github-linguist/linguist/assets/120581772/d7e47640-fd5d-4c1d-8566-7c8c0c8ca234)

**Comment by lildude** [comment]
This is happening because your repo contains files with a `.php` extension that don't actually contain any PHP. As a result, the [heuristic](https://github.com/github-linguist/linguist/blob/4ac734c15a96f9e16fd12330d0cb8de82274f700/lib/linguist/heuristics.yml#L527-L532) we use to determine PHP vs Hack isn't finding anything that matches either language so we're falling through to the classifier which is making a guess based on the content of the samples we have.

There are three solutions:

1. Change the extension to reflect the actual content, in this case `.html` - this will probably improve the syntax highlighting too
2. Add PHP code to these files
3. Implement an [override](https://github.com/github-linguist/linguist/blob/master/docs/overrides.md) to force the language of these files
